core/shared/src/main/scala/zio/duration/Duration.scala

```diff
@@ -104,6 +106,16 @@ object Duration {
     override def asScala: ScalaDuration = ScalaFiniteDuration(nanos, TimeUnit.NANOSECONDS)
 
     override def asJava: JavaDuration = JavaDuration.ofNanos(nanos)
+
+    override def render: String = toMillis match {
```

Comment:
Are we okay with losing so much precision here? If a duration is 1500 milliseconds for example that is going to get displayed as 1 second. Seems like there are some use cases where that could be okay but others where it could be very undesirable. Maybe for values in between we divide by `1000.0` or `60000.0` and report the double value plus the unit, possibly with some formatting of the double value so it doesn't have too many digits?

Reply:
I didn’t change how it was but yeah maybe we can do better. I’ll have a look at how it’s done elsewhere.

Reply:
You're right. I messed up [here](https://github.com/zio/zio/blob/e83b74cfef73cbf952a6bec5418d92855a92f1f1/test/shared/src/main/scala/zio/test/TimeoutVariants.scala#L80) by doing integral instead of fractional division. I'm just thinking this is what is going to show up for anyone who is using `Duration` for anything. So there is a lot of opportunity for improvement since seeing the nanoseconds today is not that useful but we want to make sure we don't cause problems for people who are doing different things with time.

Reply:
I improved it as follows:
```
4 ms
3 s
3 s 459 ms
2 min
2 min 43 s
```
I'm only displaying 1 unit below except when it's zero.